examples/dgfip_c/ml_primitif/Makefile

```diff
@@ -1,16 +1,287 @@
-include ../../../Makefile.include
+ifeq ($(MAKELEVEL), 0)
```

Comment:
Yes, this is the point.

MAKELEVEL variable is incremented by recursive calls to Make. 0 is for a direct call from the command line. If the Makefile is read and executed from a $(MAKE) call in the root Makefile, MAKELEVEL is 1. If itself it calls another sub-Makefile with $(MAKE), it'll pass the value MAKELEVEL=2 to it.

It allows to only load the Makefile.include (and the nested Makefile.config) once at the command line call level, so every sub-make can be independently called and use Makefile.include, or be called from a Makefile above and use the inclusion from the caller.

In order to do this trick, needed variables must be exported in Makefile.include to make them available in the sub-make environments.

I've done this because it allows us to use functions and other side-effect-heavy Make constructions in the config file, which should be executed only once per call. For example, I directly drive the DGFiP SVN system from the Makefile.config, so if its content were executed twice, it simply would waste time and resources cloning twice or updating twice the repository.